src/Redux/UserReducer.js

```diff
@@ -0,0 +1,110 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { toast } from 'react-toastify';
+
+const notify = (e) => toast(e);
+
+const UserApi = 'http://localhost:3000/api/v1/users';
+
+// fetch all users for smoother login
+export const fetchUsers = createAsyncThunk('user/getUsers', async () => {
+  const res = await fetch(UserApi);
+  const data = res.json();
+  const users = await data;
+  return users;
+});
+
+// Create user on sign up params {name: ''}
+export const fetchCreateUser = createAsyncThunk('user/signup', async (user) => fetch(UserApi, {
+  method: 'POST',
+  headers: {
+    'Content-type': 'application/json',
+  },
+  body: JSON.stringify(user),
+}).then(async (res) => {
+  const data = res.json();
+  const user = await data;
+  return user;
+}));
+
+const options = {
+  name: 'User',
+  initialState: {
+    users: [],
+    currentUser: {},
+    error: '',
+    pending: false,
+    rejected: false,
+    fulfilled: false,
+  },
+  reducers: {
+    setCurrentUser(state, action) {
+      const currentUserState = state;
+      currentUserState.currentUser = action.payload;
+    },
+    signOut(state) {
+      // rerender the login page.
+      // remove current user.
+      const currentUserState = state;
+      currentUserState.currentUser = {};
+      localStorage.removeItem('currentUser');
+      // notify the user that he was signed out
+      notify('signed out');
+      // reload the page to render authentication.
+      window.location.reload();
+    },
+  },
```

Comment:
[OPTIONAL] Great work on the login functionality. Kindly implement the logout functionality as well because once logged in the user is locked in the system and cannot end the session.

Reply:
It was working on my side did you test it?